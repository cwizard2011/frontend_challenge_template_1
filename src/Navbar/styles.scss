.container {
  display: flex;
}

.wrapper {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%;
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
}

.deals {
  display: flex;
}

.fontAwesome {
  font-family: FontAwesome;
}

.no-product-found {
  width: 100%;
  display: flex;
  justify-content: center;
  
  & p {
    font-family: 'Montserrat';
    margin-top: 2px;
    font-size: 25px
  }
}

.search-bar-style {
  margin-top: 5px;
  border-radius: 20px;
  width: 500px;
  margin-right: 80px;
  padding-left: 20px;
  outline: none !important;
}

.required-field-style {
  color: red;
}

@media only screen and (max-width: 767px) {
  .search-bar-style {
    width: 200px;
  }
}
