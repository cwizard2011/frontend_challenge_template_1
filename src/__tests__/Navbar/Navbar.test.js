import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Navbar from '../../Navbar';

describe('<Navbar />', () => {

  const props = {
    productIncart: [],
    cartPrice: '0.00',
    searchProducts: [],
    searchTerm: '',
    history: {
      location: {
        pathname: ''
      }
    },
    setDepartment: jest.fn(),
    department: ''
  }
  it('renders without crashing', () => {
    const wrapper = shallow(<Navbar {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});