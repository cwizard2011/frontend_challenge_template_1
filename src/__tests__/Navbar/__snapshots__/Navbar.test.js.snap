// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders without crashing 1`] = `
<div>
  <Navbar
    expand="md"
    light={true}
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    tag="nav"
  >
    <NavbarToggler
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <p
          style={
            Object {
              "fontFamily": "Montserrat",
              "marginLeft": 20,
              "marginTop": 8,
            }
          }
        >
          Hi
        </p>
        <NavItem
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
          tag="li"
        >
          <NavLink
            onClick={[Function]}
            style={
              Object {
                "color": "#f7436b",
                "cursor": "pointer",
                "fontFamily": "Montserrat",
                "fontWeight": "bold",
              }
            }
            tag="a"
          >
            Sign in
          </NavLink>
        </NavItem>
        <NavItem
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
          tag="li"
        >
          <p
            style={
              Object {
                "fontFamily": "Montserrat",
                "marginTop": 8,
              }
            }
          >
            Or
          </p>
          <NavLink
            onClick={[Function]}
            style={
              Object {
                "color": "#f7436b",
                "cursor": "pointer",
                "fontFamily": "Montserrat",
                "fontWeight": "bold",
              }
            }
            tag="a"
          >
            Register
          </NavLink>
        </NavItem>
      </Nav>
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          style={
            Object {
              "marginRight": 20,
            }
          }
          tag="li"
        >
          <NavLink
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "flex",
              }
            }
            tag="a"
          >
            <i
              className="fas fa-shopping-bag"
              style={
                Object {
                  "color": "black",
                  "fontSize": 20,
                  "position": "absolute",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#f7436b",
                  "borderRadius": "50%",
                  "color": "#FFF",
                  "height": 20,
                  "marginLeft": 10,
                  "marginTop": -10,
                  "position": "absolute",
                  "textAlign": "center",
                  "width": 20,
                }
              }
            >
              <p
                style={
                  Object {
                    "fontFamily": "Montserrat",
                    "fontSize": 14,
                    "marginTop": 0,
                  }
                }
              >
                0
              </p>
            </div>
          </NavLink>
        </NavItem>
        <NavItem
          tag="li"
        >
          <NavLink
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "fontFamily": "Montserrat",
              }
            }
            tag="a"
          >
            Your Bag: 
          </NavLink>
        </NavItem>
        <NavItem
          tag="li"
        >
          <NavLink
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "fontFamily": "Montserrat",
              }
            }
            tag="a"
          >
            $ 
            0.00
             
          </NavLink>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
  <Navbar
    expand="md"
    light={true}
    style={
      Object {
        "backgroundColor": "black",
      }
    }
    tag="nav"
  >
    <NavbarToggler
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      style={
        Object {
          "justifyContent": "space-between",
        }
      }
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
          tag="li"
        >
          <NavLink
            href="/"
            style={
              Object {
                "color": "#f7436b",
                "fontFamily": "Montserrat",
                "fontSize": 20,
              }
            }
            tag="a"
          >
            SHOPMATE
          </NavLink>
        </NavItem>
      </Nav>
      <Nav
        className=""
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          style={
            Object {
              "marginRight": 20,
            }
          }
          tag="li"
        >
          <NavLink
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "flex",
              }
            }
            tag="a"
          >
            <i
              className="fas fa-shopping-bag"
              style={
                Object {
                  "color": "#FFF",
                  "fontSize": 20,
                  "position": "absolute",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#f7436b",
                  "borderRadius": "50%",
                  "color": "#FFF",
                  "height": 20,
                  "marginLeft": 10,
                  "marginTop": -10,
                  "position": "absolute",
                  "textAlign": "center",
                  "width": 20,
                }
              }
            >
              <p
                style={
                  Object {
                    "fontFamily": "Montserrat",
                    "fontSize": 14,
                    "marginTop": 0,
                  }
                }
              >
                0
              </p>
            </div>
          </NavLink>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;
